src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -517,33 +517,44 @@ private void stopConnection(String errorMessage) {
         RuntimeException exception = null;
         this.state.lock();
         try {
+            ConnectionState connectionState = this.state.getConnectionStateUnsynchronized(true);
+
+            if (connectionState == null)
+            {
+                this.logger.error("'stopConnection' called with a null ConnectionState. This is not expected, please file a bug.");
```

Comment:
Let's add a reference to https://github.com/dotnet/aspnetcore/issues/new?assignees=&labels=&template=bug_report.md in this.